core/shared/src/main/scala/spire/math/Interval.scala

```diff
@@ -767,6 +767,48 @@ sealed abstract class Interval[A](implicit order: Order[A]) extends Serializable
 
   def foldOver[B](init: B, step: A)(f: (B, A) => B)(implicit ev: AdditiveMonoid[A], nt: NumberTag[A]): B =
     iterator(step).foldLeft(init)(f)
+
+  /**
+    * Result of overlapping this interval with another one.
+    * Can be one of the following:
+    * - [[spire.math.interval.Equals]] if intervals are equal
+    * - [[spire.math.interval.StrictlyLess]] if intervals are notEmpty don't intersect
```

Comment:
It seems a bit awkward that, eg, `lhs.overlap(rhs)` would return `StrictlyLess`, if `rhs` is strictly less than `lhs`. I wonder if maybe a better name would be something like `Disjoint`, instead of `StrictlyLess`. The order of the parameters in `Disjoint` would still maintain the order (eg 1st arg < 2nd arg), but the name seems to match the intent of the result a bit better.